src/main/java/br/com/deveficiente/bolaoapi/shared/validator/EmailCollectionValidator.java

```diff
@@ -0,0 +1,18 @@
+package br.com.deveficiente.bolaoapi.shared.validator;
+
+import org.hibernate.validator.internal.constraintvalidators.bv.EmailValidator;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import java.util.Set;
+
+public class EmailCollectionValidator implements ConstraintValidator<EmailCollection, Set<String>> {
+
+    public boolean isValid(Set<String> emails, ConstraintValidatorContext context) {
+        if (emails == null) return false;
+
+        EmailValidator validator = new EmailValidator();
```

Comment:
*Usamos tudo que conhecemos que está pronto. Só fazemos código do zero se for estritamente necessário.*

Ótimo aqui. Por isso que é importante conhecer as tecnologias que usamos :). Reaproveitamento massa de código!

Reply:
Pena que a classe vai morrer :P

Reply:
Pena? 
Foi uma beleza, agora não tenho que criar mais validations para coleções.
Nunca mais na vida!